source/extensions/tracers/xray/tracer.cc

```diff
@@ -0,0 +1,124 @@
+#include "extensions/tracers/xray/tracer.h"
+
+#include <chrono>
+#include <ctime>
+#include <random>
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/xray/util.h"
+#include "extensions/tracers/xray/xray_core_constants.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+const std::string Tracer::VERSION_ = "1";
+const std::string Tracer::DELIMITER_ = "-";
+
+std::string Tracer::generateRandom96BitString() {
+  char values[25] = {'\0'};
+  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+  std::mt19937 mt_rand(seed);
+
+  for (int i = 0; i < 24; i++) {
+    values[i] = hex_digits[mt_rand() % 16];
+  }
+  return values;
+}
+
+std::string Tracer::generateTraceId() {
+  time_t epoch = time(nullptr);
```

Comment:
Use TimeSource.getSystemTime() here rather than directly calling the C library.

Also, does it matter that lots of threads may wind up with the same trace ID?

Reply:
I used time_source.systemTime() instead. For answering your question, trace id is unique and different and that is why we use a 24 hexadecimal random digits to differentiate it.

Reply:
Yes that's what I meant; `time_source.systemTime()` :)